test/integration/main_test.go

```diff
@@ -32,24 +35,84 @@ func KubeClient() (result *kubernetes.Clientset) {
 		os.Exit(1)
 	}
 
-	kubeClient, err := kubernetes.NewForConfig(config)
+	clientset, err := kubernetes.NewForConfig(config)
 	if err != nil {
 		panic(err.Error())
 	}
-	return kubeClient
+	return clientset
 }
 
-func RestartInsightsOperator(t *testing.T) {
+func configV1Client() (result *configv1client.ConfigV1Client) {
+	kubeconfig, ok := os.LookupEnv("KUBECONFIG") // variable is a path to the local kubeconfig
+	if !ok {
+		fmt.Printf("kubeconfig variable is not set\n")
+	} else {
+		fmt.Printf("KUBECONFIG=%s\n", kubeconfig)
+	}
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		fmt.Printf("%#v", err)
+		os.Exit(1)
+	}
+
+	client, err := configv1client.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+	return client
+}
+
+func clusterOperatorInsights() *configv1.ClusterOperator {
+	// get info about insights cluster operator
+	operator, err := configClient.ClusterOperators().Get("insights", metav1.GetOptions{})
+	if err != nil {
+		panic(err.Error())
+	}
+	return operator
+}
+
+func isOperatorDegraded(t *testing.T, operator *configv1.ClusterOperator) bool {
+	statusConditions := operator.Status.Conditions
+
+	for _, condition := range statusConditions {
+		if condition.Type == "Degraded" {
+			if condition.Status != "True" {
+				t.Log("Operator is not degraded")
+				return false
+			}
+		}
+	}
+	t.Log("Operator is degraded")
+	return true
+}
+
+func isOperatorDisabled(t *testing.T, operator *configv1.ClusterOperator) bool {
+	statusConditions := operator.Status.Conditions
+
+	for _, condition := range statusConditions {
```

Comment:
this might be refactored as the code is almost the same as in `isOperatorDegraded`.

Reply:
but ok for tests :)